crates/ide/src/hover.rs

```diff
@@ -111,100 +113,179 @@ pub(crate) fn hover(
         kind if kind.is_trivia() => 0,
         _ => 1,
     })?;
-    let token = sema.descend_into_macros(token);
-    let node = token.parent()?;
+
+    let mut seen = HashSet::default();
+
+    let mut fallbacks = vec![];
```

Comment:
Since we only want the first/one fallback we can probably replace this with an `Option` that only gets populated once.

Reply:
agree - thought about that and then had a brain fart that told me it wouldn't work out. Will fix.

Reply:
oh, actually there is a problem with that after all - `type_hover` might also return `None`. But I suppose that's ok, just keep trying till we get `Some`